eth-contracts/contracts/CKBChain.sol

```diff
@@ -69,6 +73,33 @@ contract CKBChain is ICKBChain, ICKBSpv {
 
     /// #ICKBSpv
     function proveTxExist(bytes calldata txProofData, uint64 numConfirmations) external view returns (bool){
+        bytes29 proofView = txProofData.ref(uint40(ViewSpv.SpvTypes.CKBTxProof));
+        uint64 blockNumber = proofView.spvBlockNumber();
+        bytes32 blockHash = proofView.blockHash();
+        require(blockNumber + numConfirmations <= lastBlockNumber, "blockNumber is too ahead");
+        require(canonicalHeaderHashes[blockNumber] == blockHash, "blockNumber and blockHash mismatch");
+        require(canonicalTransactionRoots[blockHash] != bytes32(0), "blockHash invalid or too old");
```

Comment:
the logic can not support the situation: `numConfirmations < 500`

Reply:
If Locker/other service do not need a higher safety rating, they could provide a smaller numConfirmations for efficiency

Reply:
the line 80 may panic when the numConfirmations is smaller than 500